src/main/java/dog/pawbook/model/dog/DateOfBirth.java

```diff
@@ -0,0 +1,75 @@
+package dog.pawbook.model.dog;
+
+import static dog.pawbook.commons.util.AppUtil.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.Period;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.Date;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Represents a Dog's breed in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidBreed(String)}
+ */
+public class DateOfBirth {
+
+    public static final String MESSAGE_CONSTRAINTS = "Age should only contain numbers and should not be blank.";
+    public static final String VALIDATION_REGEX = "^(3[01]|[12][0-9]|0[1-9])-(1[0-2]|0[1-9])-[0-9]{4}$";
+
+    public final String value;
+
+    /**
+     * Construct an {@code DateOfBirth}.
+     *
+     * @param dob A valid date of birth (dob).
+     */
+    public DateOfBirth(String dob) {
+        requireNonNull(dob);
+        checkArgument(isValidDob(dob), MESSAGE_CONSTRAINTS);
+        value = dob;
+    }
+
+    /**
+     * Returns true if a given string is a valid email.
+     */
+    public static boolean isValidDob(String test) {
+        Pattern pattern = Pattern.compile(VALIDATION_REGEX);
+        Matcher matcher = pattern.matcher(test);
+        return matcher.matches();
+    }
+
+    public int getAge(String dob) throws ParseException {
+        SimpleDateFormat formatter = new SimpleDateFormat("dd-MM-yyyy");
+        Date date = formatter.parse(dob);
+        //Converting obtained Date object to LocalDate object
+        Instant instant = date.toInstant();
+        ZonedDateTime zone = instant.atZone(ZoneId.systemDefault());
+        LocalDate givenDate = zone.toLocalDate();
+        Period period = Period.between(givenDate, LocalDate.now());
```

Comment:
Is it cleaner if we just parse using `LocalDate` right from the start? Something like `LocalDate localDate = LocalDate.parse(dob, DateTimeFormatter.ofPattern("d-M-yyyy"));`